src/main/java/com/music/tagger/controller/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.music.tagger.controller;
+
+import com.music.tagger.exceptions.TrackNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@Slf4j
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({TrackNotFoundException.class, IllegalStateException.class})
+    public ResponseEntity<Object> handleInternal(Exception ex, WebRequest request) {
+        log.warn("406 Status Code", ex);
```

Comment:
406 is not   HttpStatus.INTERNAL_SERVER_ERROR(500)!
Why TrackNotFoundException.class match to 500?

Reply:
This is from a copied block of code. I forgot to change it when I chose the response code that best fits this method.